src/projections/analysis/StsReader.java

```diff
@@ -172,9 +174,10 @@ public StsReader(String FileName)
 		    Machine = matchQuotes(st);
 		} else if (s1.equals("PROCESSORS")) {
 		    NumPe = Integer.parseInt(st.nextToken());
-		} else if (s1.equals("SMPMODE")) {
-		    NodeSize = Integer.parseInt(st.nextToken());
- 		    NumNodes = Integer.parseInt(st.nextToken());
+		} else if (s1.equals("SMPMODE")) { 
+			NodeSize = Integer.parseInt(st.nextToken());
+			NumNodes = Integer.parseInt(st.nextToken());
+			isSMP = true;
```

Comment:
Would this change cause `isSMPRun()` to always return true? Is there a point to keeping that flag in that case? Maybe instead it should be true if there is a comm thread or if ppn > 1.

Reply:
No, the `SMPMODE` field will only exist in the STS when using an SMP version of Charm++, so `isSMP` will be false when loading a trace corresponding to a program that was built with a non-SMP build of Charm++.

I think Multicore should probably have `isSMP` set to false, but that's a change that should be made on the tracing side, not the Projections side.